x/builder/ante/ante.go

```diff
@@ -96,6 +96,27 @@ func (ad BuilderDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate bool,
 		if err := ad.builderKeeper.ValidateBidInfo(ctx, topBid, bidInfo); err != nil {
 			return ctx, errors.Wrap(err, "failed to validate auction bid")
 		}
+
+		// Check that all bundle transactions are valid.
+		for _, txBz := range bidInfo.Transactions {
+			tx, err := ad.txDecoder(txBz)
+			if err != nil {
+				return ctx, errors.Wrap(err, "failed to decode bundled transaction")
+			}
+
+			// Check if the transaction has already been accepted into the mempool.
+			contains, err := ad.mempool.Contains(tx)
+			if err != nil {
+				return ctx, errors.Wrap(err, "failed to check if transaction is in mempool")
+			}
+
+			// If the transaction is not in the mempool, we need to validate it.
+			if !contains {
+				if ctx, err = next(ctx, tx, simulate); err != nil {
```

Comment:
Is `ctx, err = next(ctx, tx, simulate)` going to sufficiently apply state changes to `ctx` for all bundled transactions? Or does `next` already apply the changes to current `ctx`?

Reply:
We want state changes applied in the order of transactions of the bundle. Just want to make sure this does that.